File: day1/part1.js
startTimer = new Date();
const input = require('fs').readFileSync('input.txt', 'utf8').split('\n');

let sum = 0;
for (let i = 0; i < input.length; i++) {
    let left, right
    for (let c = 0; c < input[i].length; c++) {
        if (input[i][c] >= '0' && input[i][c] <= '9') {
            if (left === undefined) {
                left = input[i][c];
                right = input[i][c];
            } else {
                right = input[i][c];
            }
        }

    }
    sum += parseInt(left) * 10 + parseInt(right);
}
console.log(sum);
endTimer = new Date();
console.log(endTimer - startTimer + " ms");

File: day1/part1.scala
object Part1 {
  def main(args: Array[String]): Unit = {
    import scala.io.Source
    import System.currentTimeMillis

    val startTimer = currentTimeMillis()
    val data = Source.fromFile("input.txt").getLines().toList

    var sum = 0
    for (line <- data) {
      var left, right: Option[Char] = None
      for (i <- line.indices) {
        if (line(i).isDigit) {
          if (left.isEmpty) {
            left = Some(line(i))
            right = Some(line(i))
          } else {
            right = Some(line(i))
          }
        }
      }
      sum += (left.get.asDigit * 10) + right.get.asDigit
    }
    println(sum)
    println(s"${(currentTimeMillis() - startTimer)} ms")
  }
}